pages/feed/index.tsx

```diff
@@ -85,6 +85,40 @@ const Feeds: WithPageLayout<HighlightSSRProps> = (props: HighlightSSRProps) => {
     { name: "Highlights", count: highlights_count ?? 0 },
   ];
 
+  useEffect(() => {
+    const queryParams = new URLSearchParams(window.location.search);
+    const newHighlight = queryParams.get("new");
+    const signInRequired = queryParams.get("signIn");
+
+    if (newHighlight && signInRequired) {
+      signIn({ provider: "github", options: { redirectTo: `${window.location.origin}/feed?new=${newHighlight}` } });
+    }
+    var focusOnHighlighCreationInput: NodeJS.Timeout;
+
+    if (window.innerWidth > 768) {
+      focusOnHighlighCreationInput = setInterval(() => {
+        const highlightCreationInput = document.getElementById("highlight-create-input");
+        if (newHighlight && highlightCreationInput) {
+          highlightCreationInput.click();
+          highlightCreationInput.focus();
+          clearInterval(focusOnHighlighCreationInput);
+        }
+      }, 1000);
+    } else {
+      // for mobile. No need to focus on input, just click on the button as it opens up a form anyway.
+      focusOnHighlighCreationInput = setInterval(() => {
+        const mobileHighlightCreateButton = document.getElementById("mobile-highlight-create-button");
+        if (newHighlight && mobileHighlightCreateButton) {
+          mobileHighlightCreateButton.click();
+          clearInterval(focusOnHighlighCreationInput);
```

Comment:
If the ?new isn't passed on desktop, won't this keep checking forever? Why isn't this a one-time check?

Reply:
Yeah I should be creating these intervals only when ?new is passed, I'll change that.

Regarding why this is not a one time check,
I just didn't get one of these elements without waiting for some time (it just doesn't get added to the DOM before this line of code executes).